service/src/java/org/apache/hive/service/CookieSigner.java

```diff
@@ -32,7 +32,7 @@
  */
 public class CookieSigner {
   private static final String SIGNATURE = "&s=";
-  private static final String SHA_STRING = "SHA-256";
+  private static final String SHA_STRING = "SHA-512";
   private byte[] secretBytes;
   private static final Logger LOG = LoggerFactory.getLogger(CookieSigner.class);
 
```

Comment:
Does the change have backward compatible issues?

Reply:
No. If the cookie doesn't match, it would ask for credentials for the first time or send the newly generated cookie.